spec.md

```diff
@@ -537,35 +537,33 @@ message VolumeInfo {
   // NFS share). If set, it MUST be non-zero.
   uint64 capacity_bytes = 1;
 
-  // Contains identity information for the created volume. This field is  
+  // Contains identity information for the created volume. This field is
   // REQUIRED. The identity information will be used by the CO in
   // subsequent calls to refer to the provisioned volume.
-  VolumeID id = 4;
-
-  // Metadata of the created volume. This field is OPTIONAL. If set, the
-  // CO SHALL pass this information along with the `id` to subsequent
-  // calls.
-  VolumeMetadata metadata = 5;
+  VolumeHandle handle = 2;
 }
 
-// The identity of the volume.
-message VolumeID {
-  // The identity of the provisioned volume specified by the Plugin in
-  // the form of key-value pairs. This field is REQUIRED. Given this
-  // information will be passed around by the CO, it is RECOMMENDED that
-  // each Plugin keeps this information as small as possible.
-  map<string, string> values = 1;
-}
-
-// The metadata information about the volume.
-message VolumeMetadata {
-  // The metadata information about the provisioned volume specified by
-  // the Plugin in the form of key-value pairs. This field is OPTIONAL.
-  // This field MAY contain sensitive information. Therefore, the CO
-  // MUST NOT leak this information to untrusted entities. Given this
-  // information will be passed around by the CO, it is RECOMMENDED that
-  // each Plugin keeps this information as small as possible.
-  map<string, string> values = 1;
+// VolumeHandle objects are generated by Plugin implementations to
+// serve two purposes: first, to efficiently identify a volume known
+// to the Plugin implementation; second, to capture additional, opaque,
+// possibly sensitive information associated with a volume. It SHALL
+// NOT change over the lifetime of the volume and MAY be safely cached
+// by the CO.
+// Since this object will be passed around by the CO, it is RECOMMENDED
+// that each Plugin keeps the information herein as small as possible.
+message VolumeHandle {
+
```

Comment:
nit: kill this line to be consistent with other places.

Reply:
ack